mleap-xgboost-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/mleap/XGBoostClassificationModelParitySpec.scala

```diff
@@ -33,6 +33,8 @@ class XGBoostClassificationModelParitySpec extends SparkParityBase {
   // These params are not needed for making predictions, so we don't serialize them
   override val unserializedParams = Set("labelCol", "evalMetric", "objective")
 
+  override val excludedColsForComparison = Array[String]("prediction")
```

Comment:
Exclude this column when comparing result.
The reason is xgboost might generate some prediction result with equal probability ( 0.5/0.5 ) on each label, then the prediction label might be nondeterministic 0 or 1.
So I exclude the prediction column.
But, we still compare the "probability" column which can ensure the model inference parity.

Reply:
If we set the random seed can we get things to be deterministic for the test?

Reply:
Seemingly no. There're some other non-deterministic factors such as multiple spark tasks run in parallelism and causes different sum order when aggregation.

Assuming it works, I still don't recommend the approach because next time xgboost upgrading we might need to update test again.

Reply:
😢 but makes sense